src/main/java/com/synopsys/integration/detect/workflow/blackduck/developer/blackduck/RapidScanConfigBdio2StreamUploader.java

```diff
@@ -58,7 +58,14 @@ public HttpUrl start(BdioFileContent header, BlackDuckRequestBuilderEditor edito
             .apply(editor)
             .buildBlackDuckResponseRequest(url);
         Response response = recursiveExecute(request, 0L, 0, detectTimeout);
-        HttpUrl responseUrl = new HttpUrl(response.getHeaderValue("location"));
+        String location = response.getHeaderValue("location");
+        
+        if (location == null) {
+            throw new IntegrationException("Attempted to create a scan by uploading to " + url.toString()
+                    + " but failed with " + response.getStatusCode() + " response code.");
```

Comment:
Would it be more useful to mention that `location` is missing rather than reporting the status code (or maybe in addition to)?

Reply:
Will the user have any recourse in this situation in terms of being able to rectify the issue themselves via modifying the configuration provided for the Detect run, or that they could provide information to an admin on the BD/IT side to aid in resolving the issue? (Always nice to be able to provide either some guidance, like "... verify that the provided XYZ configuration value provided at runtime is correct", so they can resolve the issue, or that can be provided to someone that can.)

Reply:
I don't think the user can do much here sadly. In this scenario we are trying to hit BlackDuck's /api/developer-scans endpoint to start a scan. 9/10 that should just work...when it doesn't it means something is wrong with BlackDuck and unfortunately in most scenarios not much more than the status code comes back to tell us what happened.

Reply:
Ya, figured that was going to be the case. Perhaps include the info for the location in case it assists BD troubleshooting.
To make it extra clear that the issue is on the BD side, maybe add ..." response code **from the Black Duck server**.");